acvm-repo/blackbox_solver/src/bigint.rs

```diff
@@ -51,6 +65,18 @@ impl BigIntSolver {
         modulus: &[u8],
         output: u32,
     ) -> Result<(), BlackBoxResolutionError> {
+        if self.pedantic_solving
+            && (!self.is_valid_modulus(modulus) || inputs.len() > modulus.len())
+        {
+            if !self.is_valid_modulus(modulus) {
+                panic!("--pedantic-solving: bigint_from_bytes: disallowed modulus {:?}", modulus);
+            } else {
+                panic!(
+                    "--pedantic-solving: bigint_from_bytes: inputs.len() > modulus.len() {:?}",
+                    (inputs.len(), modulus.len())
+                );
+            }
+        }
```

Comment:
```suggestion
        if self.pedantic_solving {
            if !self.is_valid_modulus(modulus) {
                panic!("--pedantic-solving: bigint_from_bytes: disallowed modulus {:?}", modulus);
            }
            if inputs.len() > modulus.len() {
                panic!(
                    "--pedantic-solving: bigint_from_bytes: inputs.len() > modulus.len() {:?}",
                    (inputs.len(), modulus.len())
                );
            }
        }
```
When I saw just the inner `if` I thought we would panic either way. I think this is cleaner because:
* We only check `is_valid_modulus` once
* The check on `inputs` vs `modulus` is close to the panic, more obvious